### PR TITLE
Fix/ TestTendermintBufferFailing on windows

### PR DESCRIPTION
**Pull request by 0xArdi**
## Proposed changes

~~Update the skip reason.~~
Fixes `TestTendermintBufferFailing` on windows. 

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: tests/test_deployments/test_app.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def setup_class(cls) -> None:
 
     @pytest.mark.skipif(
         platform.system() == "Windows",
-        reason="Needs investigation. Could be: Tendermint process is not killed on teardown in Windows.",
+        reason="The pipe gets filled before the test reaches here, and the node becomes unresponsive.",
```
Comment (0xArdi) [inline tests/test_deployments/test_app.py:313]: We can make this work by having a different fixture instead of `@wait_for_node_to_run`.

Comment (DavidMinarsch) [reply tests/test_deployments/test_app.py:313]: Yes please make it work.

File: tests/test_deployments/conftest.py [hunk 2]
```diff
@@ -76,3 +76,17 @@ def wait_for_node(
...
     assert success, "Tendermint node not running"
     yield
+
+
+@pytest.fixture
+def wait_for_occupied_rpc_port(
+    http_: str, loopback: str, rpc_port: int
+) -> Generator[None, None, None]:
+    """Wait for Tendermint to occupy rpc_port."""
+    i, max_retries = 0, 5
```
Comment (DavidMinarsch) [inline tests/test_deployments/conftest.py:86]: Make max_retries configurable please. Also 5 seems very low for CI...

File: tests/test_deployments/conftest.py [hunk 3]
```diff
@@ -76,3 +76,17 @@ def wait_for_node(
...
+    while not __port_is_open(loopback, rpc_port):
+        logging.debug(f"waiting for node... t={i}")
+        i += 1
+        time.sleep(1)
```
Comment (DavidMinarsch) [inline tests/test_deployments/conftest.py:90]: would make sleep time also configurable ... just makes the fixture more useful?

Comment (0xArdi) [reply tests/test_deployments/conftest.py:90]: copied from fixture above, which was working until now

File: tests/test_deployments/conftest.py (hunk 2 above)
Comment (0xArdi) [reply tests/test_deployments/conftest.py:86]: copied from above

**Review by Adamantios — APPROVED** [review]

File: tests/test_deployments/conftest.py [hunk 4]
```diff
@@ -64,15 +76,29 @@ def __port_is_open(ip: str, port: int) -> bool:
...
+    http_: str, loopback: str, rpc_port: int, max_retries: int, sleep_amount: int
+) -> Generator[None, None, None]:
+    """Wait for Tendermint to occupy rpc_port."""
+    i = 0
+    while not __port_is_open(loopback, rpc_port):
+        logging.debug(f"waiting for node... t={i}")
+        i += 1
+        time.sleep(sleep_amount)
+        assert i < max_retries, "Failed to run tendermint node in time."
+    yield
```
Comment (Adamantios) [inline tests/test_deployments/conftest.py:104]: I think this can be used in `wait_for_node` to remove some duplication.

**Comment by Adamantios** [comment]
The APY e2e test for 4 agents fails. Looking into the logs, we noticed that:

```
2022-09-23T12:01:31.6471666Z [102816]stdout>aea.exceptions.AEAActException: An error occurred during act of behaviour valory/apy_estimation_chained_abci:0.1.0/TestBehaviour:
2022-09-23T12:01:31.6488142Z [102816]stdout>Traceback (most recent call last):
2022-09-23T12:01:31.6500558Z [102816]stdout>
2022-09-23T12:01:31.6551699Z [102819]stdout>
2022-09-23T12:01:31.6560262Z [102819]stdout>ValueError: Cannot store multiple files of type <class 'NoneType'>!Should be a dictionary of filenames mapped to their objects.
2022-09-23T12:01:31.6583857Z [102822]stdout>
2022-09-23T12:01:31.6614494Z [102822]stdout>  File "/tmp/tmpg1i_ay3m/agent_00002/vendor/valory/skills/abstract_round_abci/io_/store.py", line 75, in store
2022-09-23T12:01:31.6630147Z [102825]stdout>  File "/tmp/tmpg1i_ay3m/agent_00003/vendor/valory/skills/abstract_round_abci/io_/store.py", line 75, in store
2022-09-23T12:01:31.6655058Z [102816]stdout>  File "/tmp/tmpg1i_ay3m/agent_00000/vendor/valory/skills/abstract_round_abci/io_/store.py", line 75, in store
```

which means that it is issue #1168.

**Comment by 0xArdi** [comment]
I think that having a mechanism that handles exceptions by default in `async_act` can be quite useful.
In the ElCollectooorr most of the act methods are wrapped with try/except, and error payloads are sent in case of exceptions.
What we might have is:
- Define a default error/unhandled exception payload (or payload data) and event on round side
- Wrap `async_act` in try/except
- In case of exception, send the payload above
- Handle it appropriately on the round, in case of majority failure handle it appropriately, otherwise ignore.

**Pull request merged**